control-plane/agents/src/bin/core/controller/scheduling/volume.rs

```diff
@@ -159,6 +187,63 @@ impl AddVolumeReplica {
     }
 }
 
+fn get_restricted_nodes(
+    volume_spec: &VolumeSpec,
+    volume_group_spec: &VolumeGroupSpec,
+    registry: &Registry,
+) -> HashSet<NodeId> {
+    let mut restricted_nodes = HashSet::new();
+    if volume_spec.num_replicas == 1 {
+        let specs = registry.specs();
+        // Get the list of volumes being part of the volume group.
+        let volume_group_volumes = volume_group_spec.volumes().clone();
```

Comment:
do we need to clone here? we seem to use it by reference

Reply:
Yeah we don't. :-p